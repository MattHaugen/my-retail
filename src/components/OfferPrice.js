import React from 'react';
import styled from 'styled-components';
import colors from '../constants/colors';

const StyledOfferPrice = styled.div`
  width: 100%;
  font-family: 'Arial', sans-serif;
  font-size: 30px;
  font-weight: bold;
  margin-bottom: 40px;

  .offer-price-qualifier {
    font-size: 12px;
    font-weight: normal;
    color: ${colors.darkGray};
    margin-left: 5px;
  }
`;

function getOfferPrice(offers) {
  let OfferPrice = offers.find(offer => !!offer.OfferPrice); //True/False depending on key existence

  if (OfferPrice && OfferPrice.OfferPrice.length) {
    return OfferPrice.OfferPrice[0];
  }

  return {};
}

const OfferPrice = ({ offers }) => (
  <StyledOfferPrice>
    {getOfferPrice(offers).formattedPriceValue}
    <span className="offer-price-qualifier">{getOfferPrice(offers).priceQualifier}</span>
  </StyledOfferPrice>
)

export default OfferPrice;
