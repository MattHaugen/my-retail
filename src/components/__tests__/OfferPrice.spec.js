import React from 'react';
import OfferPrice from '../OfferPrice';
import renderer from 'react-test-renderer';

describe('OfferPrice component', () => {
  it('properly displays the offer price and qualifier', () => {
    //ARRANGE
    const inputOffers = [
      {
        OfferPrice: [
          {
            formattedPriceValue: "$139.99",
            priceQualifier: "Online Price",
          }
        ]
      },
      {
        PromoPrice: [
          {
            formattedPriceValue: "$99.99",
            priceQualifier: "Limited Time",
          }
        ]
      }
    ];

    //ACT
    const component = renderer.create(<OfferPrice offers={inputOffers} />);
    const json = component.toJSON();

    //ASSERT
    expect(json).toMatchSnapshot();
  });

  it('should display nothing if no OfferPrice', () => {
    //ARRANGE
    const inputOffers = [
      {
        PromoPrice: [
          {
            formattedPriceValue: "$99.99",
            priceQualifier: "Limited Time",
          }
        ]
      }
    ];

    //ACT
    const component = renderer.create(<OfferPrice offers={inputOffers} />);
    const json = component.toJSON();

    //ASSERT
    expect(json).toMatchSnapshot();
  });
});
