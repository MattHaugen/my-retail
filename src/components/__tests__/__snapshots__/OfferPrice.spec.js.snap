// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferPrice component properly displays the offer price and qualifier 1`] = `
<div
  className="sc-bdVaJa jqxsIb"
>
  $139.99
  <span
    className="offer-price-qualifier"
  >
    Online Price
  </span>
</div>
`;

exports[`OfferPrice component should display nothing if no OfferPrice 1`] = `
<div
  className="sc-bdVaJa jqxsIb"
>
  <span
    className="offer-price-qualifier"
  />
</div>
`;

exports[`OfferPrice component should display nothing if the no OfferPrice 1`] = `
<div
  className="sc-bdVaJa jqxsIb"
>
  <span
    className="offer-price-qualifier"
  />
</div>
`;
